pandas/tests/indexing/test_loc.py

```diff
@@ -1106,3 +1107,133 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
     tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
     tm.assert_frame_equal(df, df.loc[list(idx)])
+
+
+def test_loc_setitem_df_datetime64tz_full_column_with_index():
```

Comment:
all of this need to move to pandas/tests/extension/setitem

BUT you need to use the fixtures, so you are like writing ONE test.